### PR TITLE
Added a script for automatically generating types and methods defined in protocol.json

### PR DESCRIPTION
**Pull request by ichub**

### CODE REVIEW

File: build-tools/readme.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
+#scraper.js
```
Comment (jasta) [inline build-tools/readme.md]: Rename this file `README.md` per convention.

File: build-tools/readme.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+This script generates Java code representing a type or method defined in `protocol.json`, which can be found [here](https://code.google.com/p/chromium/codesearch#chromium/src/third_party/WebKit/Source/devtools/protocol.json&q=protocol.json&sq=package:chromium&type=cs). The script generates types recursively, keeping track of any additional types that it needs to generate.
```
Comment (jasta) [inline build-tools/readme.md]: This is way too much documentation for the generic readme.  Move this into `scraper.js` and also document it's usage in the program itself when invoked with no arguments.

This is about discoverability -- you want your documentation to be as close as possible to the actual tool or system they're using, even if it just links externally.

File: build-tools/scraper.js [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+    var paramsTypeName = className + 'Request';
+    var returnsTypeName = className + 'Response';
+
+    var result = '' +
+        '@ChromeDevtoolsMethod\r\n' +
+        'public JsonRpcResult ' + commandDef.name + '(JsonRpcPeer peer, JSONObject params) {\r\n';
+
+    if (hasParams) {
+        result += '' +
+            '\tfinal ' + paramsTypeName + ' = mObjectMapper.convertValue\r\n' +
```
Comment (rickbrew) [inline build-tools/scraper.js]: what happens with all these tab characters? does IntelliJ convert them to spaces or what?

**Comment by rickbrew** [comment]
Okay I think this looks good to me :+1:  Squash 'n merge

**Pull request merged**